src/Symfony/Component/Security/Core/Authentication/Token/SwitchUserToken.php

```diff
@@ -22,15 +24,29 @@ class SwitchUserToken extends UsernamePasswordToken
     private $originatedFromUri;
 
     /**
-     * @param string|object $user              The username (like a nickname, email address, etc.), or a UserInterface instance or an object implementing a __toString method
-     * @param mixed         $credentials       This usually is the password of the user
+     * @param UserInterface $user
      * @param string|null   $originatedFromUri The URI where was the user at the switch
      *
      * @throws \InvalidArgumentException
      */
-    public function __construct($user, $credentials, string $firewallName, array $roles, TokenInterface $originalToken, string $originatedFromUri = null)
+    public function __construct($user, /*string*/ $firewallName, /*array*/ $roles, /*TokenInterface*/ $originalToken, /*string*/ $originatedFromUri = null)
     {
-        parent::__construct($user, $credentials, $firewallName, $roles);
+        if (\is_string($roles)) {
+            // @deprecated since 5.4, deprecation is triggered by UsernamePasswordToken::__construct()
```

Comment:
Why dont we do a proper deprecation notice here? Is it because the deprecation notice will be triggered in `UsernamePasswordToken`?

Reply:
Yes, this token calls the `UsernamePasswordToken` constructor, which I've implemented to use `static::class` (to include the FQCN of the child). Also triggering one here means you'll get duplicate notices when using this token